web/packages/teleport/src/Discover/Discover.test.tsx

```diff
@@ -204,3 +208,101 @@ describe('location state', () => {
     ).not.toBeInTheDocument();
   });
 });
+
+type updateProps = {
+  resourceName?: string;
+  resourceSpecForUpdate?: ResourceSpec;
+};
+
+const renderUpdate = (props: updateProps) => {
+  const defaultPref = makeDefaultUserPreferences();
+  defaultPref.onboard.preferredResources = [Resource.WEB_APPLICATIONS];
+
+  mockUserContextProviderWith(
+    makeTestUserContext({ preferences: defaultPref })
+  );
+
+  const userAcl = getAcl();
+  const ctx = createTeleportContext({ customAcl: userAcl });
+
+  const MockComponent1 = () => {
+    const { agentMeta } = useDiscover();
+    return (
+      <>
+        {agentMeta.resourceName === 'saml2' ? agentMeta.resourceName : 'saml1'}
+      </>
+    );
+  };
+
+  const testViews: ResourceViewConfig[] = [
+    {
+      kind: ResourceKind.SamlApplication,
+      views() {
+        return [
+          {
+            title: 'MockComponent1',
+            component: MockComponent1,
+          },
+        ];
+      },
+    },
+  ];
+
+  return render(
+    <MemoryRouter
+      initialEntries={[
+        { pathname: cfg.routes.discover, state: { entity: '' } },
+      ]}
+    >
+      <TeleportContextProvider ctx={ctx}>
+        <DiscoverComponent
+          eViewConfigs={testViews}
+          isUpdateFlow={true}
+          resourceSpecForUpdate={props.resourceSpecForUpdate}
+          agentMetaForUpdate={{ resourceName: props.resourceName }}
+        />
+      </TeleportContextProvider>
+    </MemoryRouter>
+  );
+};
+
+test('update flow: renders single component based on resourceSpecForUpdatvalue', () => {
+  const resourceSpecForUpdate: ResourceSpec = {
+    name: 'Connect My Computer',
+    kind: ResourceKind.ConnectMyComputer,
+    event: null,
+    icon: 'Laptop',
+    keywords: '',
+    hasAccess: true,
+  };
+
+  renderUpdate({ resourceSpecForUpdate: resourceSpecForUpdate });
+
+  expect(screen.queryAllByTestId(ResourceKind.Server).length).toBeFalsy();
```

Comment:
Not entirely sure if this (`queryAllByTestId().length).toBeFalsy()`) is the correct way to check that resources are not rendered for the update flow. lmk if theres a better way.

Reply:
Maybe:
```ts
expect(screen.queryByTestId((ResourceKind.Server)).not.toBeInTheDocument();
```